include/aws/common/error.h

```diff
@@ -198,6 +198,7 @@ enum aws_common_error {
     AWS_ERROR_DIRECTORY_NOT_EMPTY,
     AWS_ERROR_PLATFORM_NOT_SUPPORTED,
     AWS_ERROR_INVALID_UTF8,
+    AWS_ERROR_UTF8_EXTRA_VALIDATION_FAILED,
```

Comment:
don't need this if we just let the callback raise whatever error it wants

MQTT could reuse the AWS_ERROR_INVALID_UTF8, because that still makes sense. Or it could raise its own special error with "MQTT5" in the name. Or it could raise an error about string being too long, or whatever!